src/missing.c

```diff
@@ -260,6 +260,138 @@ r_ssize list_col_equal_na(r_obj* x,
 
 // -----------------------------------------------------------------------------
 
+// [[ register() ]]
+r_obj* ffi_vec_any_missing(r_obj* x) {
+  return r_lgl(vec_any_missing(x));
+}
+
+// [[ include("missing.h") ]]
+bool vec_any_missing(r_obj* x) {
+  return vec_first_missing(x) != vec_size(x);
+}
+
+// [[ include("missing.h") ]]
+r_ssize vec_first_missing(r_obj* x) {
+  r_obj* proxy = KEEP(vec_proxy_equal(x));
+  r_ssize out = proxy_first_missing(proxy);
+  FREE(1);
+  return out;
+}
+
+static inline
+r_ssize proxy_first_missing(r_obj* proxy) {
```

Comment:
It's starting to feel like we need macros to help define this sort of polymorphic operations. Though maybe the implementations are just so slightly different that they can't be automatically generated?